src/parse.js

```diff
@@ -364,7 +381,7 @@ function parse(source, root, options) {
 
```

Comment:
"optional" and "required" should actually be error conditions under editions, idk if we need that validation layer tho (being *more* permissive than protoc might be ok?)

Reply:
sg, I'll add the restrictions.